### PR TITLE
feat: translation

### PR DESCRIPTION
**Pull request by a-m-dev**
This PR adds translation to repo

~it is dependant on #20 so that one should get in first~

**Items included:**
- [x] Translation
- [x] removing all SNTable/sn-table prefixes
- [x] add `locale:generate` and `locale:check` to pkg.json
- [x] using `StyledListItemLabel` to make text look like designs  

**Not in this PR**
- [ ] add `locale:generate` to build step [preferably another PR marked with ci]

### CODE REVIEW

**Comment by cbt1** [comment]
@mattgengqlik Hi, could you help us with the new translations strings in this PR? Need to start monitoring it and approve the `src/locale/locales/en.json`

**Comment by ashrles** [comment]
Hi @cbt1, Matt is currently away so I'll be helping you get onboarded.

To get us started please verify that @qlik-oss/product-content-media-docs-team and @qlik-oss/globalization have write access to the repo.
I'll ping the writing team to request a string review on this PR and add it to the translation pipeline once done.

**Comment by cbt1** [comment]
> Hi @cbt1, Matt is currently away so I'll be helping you get onboarded.
> 
> To get us started please verify that @qlik-oss/product-content-media-docs-team and @qlik-oss/globalization have write access to the repo. I'll ping the writing team to request a string review on this PR and add it to the translation pipeline once done.

 Thx @ashrles! They should now have access ✅

**Review by saraholvid — APPROVED** [review]
LGTM!

File: package.json [hunk 1]
```diff
@@ -37,6 +37,8 @@
     "test:unit": "jest",
     "test:watch": "jest --watch",
     "test:coverage": "jest --coverage",
+    "locale:verify": "node src/locale/tools/verify-translations.js",
```
Comment (cbt1) [inline package.json:40]: Should add locale:verify to CI. But not needed for this PR.

File: src/hooks/useTranslations/index.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { stardust } from '@nebula.js/stardust';
+import registerLocale from '../../locale/src/index';
+
+interface UseTranslationsArgs {
+  translator: stardust.Translator;
+}
+
+const useTranslations = ({ translator }: UseTranslationsArgs): stardust.Translator => {
```
Comment (cbt1) [inline src/hooks/useTranslations/index.ts:8]: This is not a hook. As it's not bound by the rules of hooks.

I think you should make it into a hook tho. As currently `registerLocale` is called everytime the HeadCellMenu component is rendered.

**Review by cbt1 — APPROVED** [review]
I think you should make `useTranslations` into an actual hook, but other then that LGTM

**Review by ashrles — APPROVED** [review]
Looks good. Repo has been added to the translation pipeline.

File: src/hooks/useTranslations/index.ts (hunk 2 above)
Comment (a-m-dev) [reply src/hooks/useTranslations/index.ts:8]: Yes, that is correct. I'll try to wrap it around `useMemo` from react

**Comment by a-m-dev** [comment]
Hi @ashrles  and @saraholvid
Thanks for approving the strings. 

Actually one thing that is missing in this PR is the actual translations to other languages. usually in like in sn-table project (for example) we were adding some one from translations group to PR and he/she was handling the translations to other language with their own commit on the PR. Do we need to add some other person who handles that commit to this pr?


Screenshot 2023-08-25 at 11 48 21

**Comment by saraholvid** [comment]
> Hi @ashrles and @saraholvid Thanks for approving the strings.
> 
> Actually one thing that is missing in this PR is the actual translations to other languages. usually in like in sn-table project (for example) we were adding some one from translations group to PR and he/she was handling the translations to other language with their own commit on the PR. Do we need to add some other person who handles that commit to this pr?
> 
> Screenshot 2023-08-25 at 11 48 21

Hi @a-m-dev, 
Not sure, I only work with the english texts. 
@ashrles ?

**Comment by a-m-dev** [comment]
@saraholvid , No problem, I asked the same question within our team and apparently you are only confirming the english strings at this step. Then in future, your team will create a new pr in this repo and add the other translations in other languages. So basically this pr is unblocked and can be merged now. 

but thanks anyway for response 🙏

**Pull request merged**